src/shipping/consignment-action-creator.ts

```diff
@@ -287,29 +323,52 @@ export default class ConsignmentActionCreator {
         };
     }
 
-    private _combineLineItems(
-        consignment: ConsignmentAssignmentRequestBody,
-        existingConsignment?: Consignment,
+    private _removeLineItems(
+        lineItems: ConsignmentLineItem[],
+        consignment: Consignment,
+        cart?: Cart
+    ): ConsignmentLineItem[] {
+        if (!cart) {
+            throw new MissingDataError(MissingDataErrorType.MissingCart);
+        }
+
+        return this._hydrateLineItems(consignment.lineItemIds, cart).map(existingItem => {
+            const sharedItem = lineItems.find(lineItem => lineItem.itemId === existingItem.itemId);
+
+            return {
+                ...existingItem,
+                quantity: existingItem.quantity - (sharedItem ? sharedItem.quantity : 0),
```

Comment:
🍹Seems more readable the other way around?

```suggestion
                quantity: sharedItem ? (existingItem.quantity - sharedItem.quantity) : existingItem.quantity,
```